tools/packaging/common/istio-iptables.sh

```diff
@@ -48,40 +48,119 @@ function usage() {
   echo '      outbound traffic (i.e. "*") is being redirected (default to $ISTIO_SERVICE_EXCLUDE_CIDR).'
   echo '  -k: Comma separated list of virtual interfaces whose inbound traffic (from VM)'
   echo '      will be treated as outbound (optional)'
+  echo '  -t: Unit testing, only functions are loaded and no other instructions are executed.'
   # shellcheck disable=SC2016
   echo ''
   # shellcheck disable=SC2016
   echo 'Using environment variables in $ISTIO_SIDECAR_CONFIG (default: /var/lib/istio/envoy/sidecar.env)'
 }
 
 function dump {
-    iptables-save
-    ip6tables-save
+echo dump
+#    iptables-save
+#    ip6tables-save
 }
 
-#
-#  Function check if argument is a valid ipv4 or ipv6 address
-#
+ipv4regexp="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
 function isValidIP() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   if [[ ${1} =~ ${ipv4regexp} ]]; then
       true
-   elif [ "$1" != "${1#*:[0-9a-fA-F]}" ]; then
+   elif isIPv6 "${1}"; then
       true
    else
       false
    fi
 }
-
 #
 # Function return true if agrument is a valid ipv4 address
 #
 function isIPv4() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   if [[ ${1} =~ ${ipv4regexp} ]]; then
       true
    else
       false
   fi
 }
+#
+# Function return true if agrument is a valid ipv6 address
+#
+ipv6section="^[0-9a-fA-F]{1,4}$"
+function isIPv6() {
+  addr="$1"
+  number_of_parts=0
+  number_of_skip=0
+  IFS=':' read -r -a addr <<< "$1"
+  for part in "${addr[@]}"; do
+    # check to not exceed number of parts in ipv6 address
+    if [[ ${number_of_parts} -ge 8 ]]; then
+        return 1
+    fi
+    if [[ ${number_of_parts} -eq 0 ]] && ! [[ ${part} =~ ${ipv6section} ]]; then
+        return 1
+    fi
+    if ! [[ ${part} =~ ${ipv6section} ]]; then
+       if ! [[ "$part" == "" ]]; then
+          return 1
+       else
+          # Found empty part, no more than 2 sections '::' are allowed in ipv6 address
+          if [[ "$number_of_skip" -ge 1 ]]; then
+             return 1
+          else
```

Comment:
same here, we can then reduce the nest level to make it more readable.

Reply:
done